diff --git a/include/eigenpy/decompositions/BDCSVD.hpp b/include/eigenpy/decompositions/BDCSVD.hpp
index 9f9eb75..231b304 100644
--- a/include/eigenpy/decompositions/BDCSVD.hpp
+++ b/include/eigenpy/decompositions/BDCSVD.hpp
@@ -32,23 +32,28 @@ struct BDCSVDVisitor
 
         .def("cols", &Solver::cols, bp::arg("self"),
              "Returns the number of columns. ")
-        .def("compute",
-             (Solver & (Solver::*)(const MatrixType &matrix)) & Solver::compute,
+        .def(
+            "compute",
+            +[](Solver &self, const MatrixType &matrix) -> Solver & {
+              return self.compute(matrix);
+            },
              bp::args("self", "matrix"),
              "Method performing the decomposition of given matrix. Computes "
              "Thin/Full "
              "unitaries U/V if specified using the Options template parameter "
              "or the class constructor. ",
              bp::return_self<>())
-        .def("compute",
-             (Solver & (Solver::*)(const MatrixType &matrix,
-                                   unsigned int computationOptions)) &
-                 Solver::compute,
+        .def(
+            "compute",
+            +[](Solver &self, const MatrixType &matrix,
+                unsigned int computationOptions) -> Solver & {
+              return self.compute(matrix, computationOptions);
+            },
              bp::args("self", "matrix", "computationOptions"),
              "Method performing the decomposition of given matrix, as "
              "specified by the computationOptions parameter.  ",
              bp::return_self<>())
         .def("rows", &Solver::rows, bp::arg("self"),
              "Returns the number of rows. ")
         .def("setSwitchSize", &Solver::setSwitchSize, bp::args("self", "s"))
 
         .def(SVDBaseVisitor<Solver>());
diff --git a/include/eigenpy/decompositions/JacobiSVD.hpp b/include/eigenpy/decompositions/JacobiSVD.hpp
index 1ea45ed..4454e5e 100644
--- a/include/eigenpy/decompositions/JacobiSVD.hpp
+++ b/include/eigenpy/decompositions/JacobiSVD.hpp
@@ -31,24 +31,28 @@ struct JacobiSVDVisitor
 
         .def("cols", &JacobiSVD::cols, bp::arg("self"),
              "Returns the number of columns. ")
-        .def("compute",
-             (JacobiSVD & (JacobiSVD::*)(const MatrixType &matrix)) &
-                 JacobiSVD::compute,
+        .def(
+            "compute",
+            +[](JacobiSVD &self, const MatrixType &matrix) -> JacobiSVD & {
+              return self.compute(matrix);
+            },
              bp::args("self", "matrix"),
              "Method performing the decomposition of given matrix. Computes "
              "Thin/Full "
              "unitaries U/V if specified using the Options template parameter "
              "or the class constructor. ",
              bp::return_self<>())
-        .def("compute",
-             (JacobiSVD & (JacobiSVD::*)(const MatrixType &matrix,
-                                         unsigned int computationOptions)) &
-                 JacobiSVD::compute,
+        .def(
+            "compute",
+            +[](JacobiSVD &self, const MatrixType &matrix,
+                unsigned int computationOptions) -> JacobiSVD & {
+              return self.compute(matrix, computationOptions);
+            },
              bp::args("self", "matrix", "computationOptions"),
              "Method performing the decomposition of given matrix, as "
              "specified by the computationOptions parameter.  ",
              bp::return_self<>())
         .def("rows", &JacobiSVD::rows, bp::arg("self"),
              "Returns the number of rows.")
 
         .def(SVDBaseVisitor<JacobiSVD>());
